### ISSUE TITLE
Discrete Class , shape function broken

### ISSUE THREAD

**Issue by aishik-pyne** [openai/gym#791]
I have upgraded gym to it's latest version.

When I run this code 
```python 
import gym
env = gym.make('MountainCar-v0')
print(env.action_space.shape)
```

I get the output 
```python
()
```

However, git cloning the repo and running it from inside the folder gives the correct output 
```
(3,)
```

Any Ideas?

**Comment by olegklimov** [comment]
It's discrete action space, it doesn't have shape, only `n`?

**Comment by aishik-pyne** [comment]
Yes but there is property function defined

```python
@property 
def shape(self):
    return (self.n,)
```

**Comment by olegklimov** [comment]
Hm... Is `n` also wrong in your setup?

**Comment by aishik-pyne** [comment]
Nope. n is working fine. 
I just found the function hence issued it. It fine :)

**Comment by bionicles** [comment]
one issue is that if we want to train across many domains, if the discrete action spaces have a different shape interface, we have to do a bunch of if statements. if it's discrete, do this, if it's not, do something else

that is less than ideal and it would be best if all environments used the same commands to retrieve the dimensions of the observation and action space. this could be streamlined A LOT

**Comment by haanvid** [comment]
Having same problem here.

**Comment by yceny** [comment]
> Yes but there is property function defined
> 
> ```python
> @property 
> def shape(self):
>     return (self.n,)
> ```

I got the same error that it gives () shape. Where should this property function defined?

**Comment by vjans** [comment]
R.I.P.

**Comment by gianmarcoaversanoenx** [comment]
I have the same issue.

**Comment by joshuap2613** [comment]
I also have the same issue

**Comment by siddhant-pradhan** [comment]
Yeah I have the same issue. In the latest version of gym, the Discrete class no longer has the shape function defined. Is there a reason why this was done/is there a workaround>

**Comment by naji-s** [comment]
wait why is this closed? the problem still exist from what i can see

**Comment by jkterry1** [comment]
PRs are welcome for this

**Comment by RedTachyon** [comment]
Is the intended behavior really `(3,)`? I think this can be "fixed" by passing the shape in the `super` call in gym/spaces/discrete.py#L17, but this would imply that the the value would be a 3-element vector. In reality, elements of a discrete space are (generally?) represented by integers, which in numpy have a shape of `()`

**Comment by naji-s** [comment]
@RedTachyon from what I can understand gym is expected to convert the discrete spaces into one-hot vectors and hence in this case the actions which are i assume are left, right, and do nothing would end up having a one-hot vector dimension of 3.
